### PR TITLE
replace tinkerpop api with neighbor accessors / scala collections

### PR DESCRIPTION
**Pull request by mpollmeier**
* replace tinkerpop sack traversal with standard for loop: Expression.parameter, MethodParameter.argument, MethodParameterOut.argument
* TrackingPointMethodsBase without tinkerpop
* some more easy refactorings away from tinkerpop
* wherever it's easily portable already, don't start traversals on node level

### CODE REVIEW

File: dataflowengine/src/main/scala/io/shiftleft/dataflowengine/language/Path.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ object Path {
 
         val trackedSymbol = trackingPoint match {
           case _: nodes.MethodParameterIn =>
-            val paramsPretty = method.parameter.l.sortBy(_.order).map(_.code).mkString(", ")
+            val paramsPretty = method.parameter.toList.sortBy(_.order).map(_.code).mkString(", ")
```
Comment (fabsx00) [inline dataflowengine/src/main/scala/io/shiftleft/dataflowengine/language/Path.scala:22]: The `.l` shorthand is pretty handy. Wonder whether we should introduce it on top of scala collections at some point. Not in this PR though.

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/MethodMethods.scala [hunk 2]
```diff
@@ -7,17 +7,17 @@ import io.shiftleft.Implicits.JavaIteratorDeco
 
 class MethodMethods(val node: nodes.Method) extends AnyVal {
 
-  def parameter: NodeSteps[nodes.MethodParameterIn] =
-    node.start.parameter
+  def parameter: Iterator[nodes.MethodParameterIn] =
+    node._methodParameterInViaAstOut
```
Comment (fabsx00) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/MethodMethods.scala:11]: I know the type is called `MethodParameterIn`, but we usually just refer to it as `parameter` throughout the rest of the code. Or is this generated?

Comment (mpollmeier) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/MethodMethods.scala:11]: `_methodParameterInViaAstOut` is generated

**Comment by mpollmeier** [comment]
> One general note: Gremlin swallows NoSuchElement exceptions, so, whenever code is ported over from Gremlin to scala iterators, we probably want to introduce exception handling to catch, report and ignore NoSuchElement.

It depends on the use case. But yes, this is an important difference when using normal collections vs gremlin.

**Pull request merged**